src/net.cpp

```diff
@@ -2808,7 +2971,10 @@ void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
         // If write queue empty, attempt "optimistic write"
         if (optimisticSend) nBytesSent = SocketSendData(*pnode);
     }
-    if (nBytesSent) RecordBytesSent(nBytesSent);
+    if (nBytesSent) {
+        RecordBytesSent(nBytesSent);
+        RecordMsgStatsSent(msg.m_type, nBytesSent, pnode->m_conn_type, pnode->ConnectedThroughNetwork());
+    }
```

Comment:
I guess this should happen unconditionally, even if nothing was sent. Just next to where `pnode->mapSendBytesPerMsgType[msg.m_type]` is updated above (and be the only place where the sent recording is done).